File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,vector<int>> nums_map;
        for(int i=0;i<nums.size();i++)
        {
            nums_map[nums[i]].push_back(i);
        }
        for(int i=0;i<nums.size();i++)
        {

            if(nums_map.count(target-nums[i])==1)
            {
                if(target-nums[i]==nums[i])
                {
                    if(nums_map[target-nums[i]].size()>1)
                    {
                        vector<int> ans;
                        ans.push_back(i);
                        ans.push_back(nums_map[target-nums[i]][1]);
                        return ans;
                    }
                    else
                    {
                        continue;
                    }
                }
                vector<int> ans;
                ans.push_back(i);
                ans.push_back(nums_map[target-nums[i]][0]);
                return ans;
            }
        }
    }
};

File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n=len(nums)
        dp = [0]*n
        dp[0]=1
        for i in range(n):
            if dp[i] ==0:
                continue
            j = 1
            while j<=nums[i] and (i+j) < n:
                dp[i+j] = 1
                j+=1
            i += nums[i]+1

        return (dp[-1]==1)
